src/main/webapp/WEB-INF/view/chat.jsp

```diff
@@ -58,15 +57,34 @@ String privacySettingButtonValue = (Boolean) request.getAttribute("isPrivate")?
     });
 
     // show contents of drop down settings menu
-    function toggleSettingsDropdown(){
-      var dropDiv = document.getElementsByClassName("dropdown-content")[0];
-      if(dropDiv.style.display === "none"){
-          dropDiv.style.display = "block";
-      }
-      else{
-          dropDiv.style.display = addUserBox.style.display = "none";
-      }
-    };
+    $(document).ready(function(){
+      $("#conversationSettings").click(function() {
+        $(".dropdown-content").toggle();
+      });
+    });
+
+    // for make private/make public button
+    var newChatPrivacyValue = "<%= privacySettingButtonValue %>";
+    $(document).ready(function() {
+      $("#privacySettingButton").click(function() {
+        fetch('/chat/<%= conversation.getTitle() %>', {
+          method: "POST",
+          headers: {
+            "Purpose": newChatPrivacyValue,
```

Comment:
What you want to use here is the `body` field of the POST request, instead of the headers. This is what is canonically used to pass the contents of the request, and it can be an arbitrary object (serialized using JSON). See: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#Body.